railties/lib/rails/generators/rails/app/interactive_options.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module Rails
+  module Generators
+    class InteractiveOptions # :nodoc:
+      include Database
+      include Thor::Shell
+
+      attr_reader :options
+
+      def self.perform(initial_options)
+        new(initial_options).perform
+      end
+
+      def initialize(initial_options)
+        @options = initial_options
+      end
+
+      def perform
+        say "Rails CLI #{Rails.version}", :green
+
+        case inquire_rails_version
+        when "dev"    then options[:dev] = true
+        when "edge"   then options[:edge] = true
+        when "main" then options[:main] = true
+        end
+
+        options[:database]             = inquire_database
+        options[:skip_action_cable]    = inquire_action_cable
+        options[:skip_action_mailer]   = inquire_action_mailer
+        options[:skip_action_mailbox]  = inquire_action_mailbox
+        options[:skip_active_storage]  = inquire_active_storage
+        options[:skip_action_text]     = inquire_action_text
+        options[:skip_bootsnap]        = inquire_bootsnap
+        options[:skip_jbuilder]        = inquire_jbuilder
+        options[:skip_javascript]      = inquire_javascript
+        options[:skip_spring]          = inquire_spring
+        options[:skip_sprockets]       = inquire_sprockets
+        options[:skip_system_tests]    = inquire_system_tests
+        options[:skip_turbolinks]      = inquire_turbolinks
+        options[:skip_webpack_install] = inquire_webpack
+        options[:webpack]              = inquire_webpack_library unless options[:skip_webpack_install]
+
+        options
+      end
+
+      private
+        def inquire_rails_version
+          say <<~MESSAGE
+          Target different versions of rails for your application.
+          *dev* will target your checked out version of rails.
+          *edge* will target the latest commit on rails.
```

Comment:
My understanding is that *edge* targets the latest stable on Rails. That's [why I created the --main option](https://github.com/rails/rails/pull/38631) to give us the latest commit (to trial out new features)

Reply:
Hmm, that is a bit confusing since the `edge` guides and API documentation point to `main`.
But i think you are correct. Looking at the tests it seems to sets the branch of rails to the stable version.
https://github.com/rails/rails/blob/a6cb4531d50b1b83b768b75dd64829521531afda/railties/test/generators/shared_generator_tests.rb#L342-L354

Reply:
Literally the story goes Eileen announced sharding in Rails. I wanted to try it out so did `rails new . --edge` and it didn't work! And there was no way to access it. So now --main gives us that. What is kind of weird is what is the purpose of `--edge` then? Is it if your local rails version is 5.2 then --edge will give you 7.0 instead of 5.2? But if you have rails latest it's the same as rails new?